scripts/extensions/videoEditor/src/VideoTimeline/VideoTimeline.tsx

```diff
@@ -0,0 +1,251 @@
+import * as React from 'react';
+import {debounce, isEmpty} from 'lodash';
+import {BarIcon} from './BarIcon';
+import {ListThumbnails} from './ListThumbnails';
+import VideoEditorContext from '../VideoEditorContext';
+import {IThumbnail} from '../interfaces';
+
+interface IProps {
+    video: React.RefObject<HTMLVideoElement>;
+    thumbnails: Array<IThumbnail>;
+    trim: {
+        start: number;
+        end: number;
+    };
+    onTrim: (start: number, end: number, runCheckIsDirty: boolean) => void;
+}
+interface IState {
+    currentTime: number;
+    thumnailsRender: Array<IThumbnail>;
```

Comment:
Because, we don't want capture thumbnails (in server) every "UI Video editor open". 
To do this, we will capture 50 (can change in settings) pictures and store them in server first. So when video editor open, we don't need capture again, we only choose a required number of thumbnails (`thumnailsRender`) in these 50 pics. 
That means `thumbnails` is data of existed 50 pictures in server. and `thumnailsRender` is a required number of thumbnails that will be rendered in UI, they are base on width of timeline (maybe 10-15 pictures) for filling up all timeline control.